keras/backend/tensorflow_backend.py

```diff
@@ -2181,6 +2181,9 @@ def get_variable_shape(x):
 
 def print_tensor(x, message=''):
     """Prints `message` and the tensor value when evaluated.
+    It therefore has to be part of the computational graph,
+    so proper usage would be like this:
+    `x = K.print_tensor(x, message="x is: ")`
```

Comment:
It's good to have an explicit example. Put it in an `# Example` section.

I don't understand the message "It therefore has to be part of the computational graph,". Please remove, the example is self-explanatory.

Reply:
The main mistake one could do would be writing 
    
    x = K.whatever(a,b,c)
    K.print_tensor(x)

Instead of 

    x = K.whatever(a,b,c)
    x = K.print_tensor(x)

That is a very common mistake (even I make it occasionally, and I understand what happens in the background). That is what "therefore it has to be part of the computational graph" was refering to. A newbie might not yet understand this.

Should one just skip that entirely, or also add this counterexample? I think it is a important point to make, and the difference is quite obvious here.

Reply:
Add something explicit like: "note that `print_tensor` returns a new tensor, which is the one you should be using in the rest of your code in order for the print operation to be taken into account."